src/Command/MailerCitizenProjectValidatedSummaryCommand.php

```diff
@@ -0,0 +1,164 @@
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $adherents = $this->adherentRepository->findBy(['status' => Adherent::ENABLED]);
+        $counter = 0;
+
+        /** @var Adherent $adherent */
+        foreach ($adherents as $adherent) {
+            // Finds all citizen projects near the Adherent
+            $citizenProjects = $this->citizenProjectRepository->findAllCitizenProjectsAroundByAdherent(
+                $adherent,
+                $adherent->getCitizenProjectCreationEmailSubscriptionRadius(),
+                self::CITIZEN_PROJECTS_LIST_LIMIT,
+                $input->getOption('maxAge')
+            );
+
+            if (0 === count($citizenProjects)) {
+                continue;
+            }
+
+            // Broadcast the citizen projects and returns them formatted in array
+            $broadcastedCitizenProjects = $this->broadcast($citizenProjects);
+
+            // Sends the message
+            $message = $this->createMessageFactory($broadcastedCitizenProjects, $adherent);
+            $this->mailer->sendMessage($message);
+
+            $this->logger->info(
+                sprintf(
+                    '[Broadcasting citizen projects] An email with %d CP has been queued for adherent #%d',
+                    count($citizenProjects),
+                    $adherent->getId()
+                )
+            );
+            ++$counter;
+        }
+        $this->logger->info(sprintf('[Broadcasting citizen projects] Total emails queued : %d', $counter));
+    }
+
+    /**
+     * Puts the citizen projects broadcast flag and returns the list.
+     *
+     * @param array $citizenProjects List of CitizenProject
+     *
+     * @return array List of citizen projects formatted for the template
+     */
+    private function broadcast(array $citizenProjects): array
+    {
+        $broadcastedCitizenProjects = [];
+
+        try {
+            $this->entityManager->beginTransaction();
+
+            /** @var CitizenProject $citizenProject */
+            foreach ($citizenProjects as $citizenProject) {
+                if ($citizenProject->isBroadcasted()) {
+                    continue;
+                }
+
+                $citizenProject->setBroadcasted(true);
+                $this->entityManager->persist($citizenProject);
+
+                array_push($broadcastedCitizenProjects, $this->boadcastedCitizenProjectFactory($citizenProject));
+            }
+
+            $this->entityManager->flush();
+            $this->entityManager->commit();
+
+            return $broadcastedCitizenProjects;
+        } catch (Exception $e) {
+            $this->logger->error(
+                '[Broadcasting citizen projects] An error occurred while broadcasting',
+                ['exception' => $e]
+            );
+        }
+    }
+
+    private function boadcastedCitizenProjectFactory(CitizenProject $citizenProject): array
+    {
+        return [
```

Comment:
Could be returned by `CitizenProject::getSummary`

Reply:
Actually this method seems useless, you should do an array query on those fields to save memory.

Reply:
Actually the `findNearByCitizenProjectSummariesForAdherent` return an array of `CitizenProject` because the `broadcast` method iterate them and put the flag.

I think a `CitizenProject::getSummary` static method is a good idea. It's ok for you ?